mesonbuild/compilers/mixins/clike.py

```diff
@@ -317,12 +317,12 @@ def _sanity_check_impl(self, work_dir: str, environment: 'Environment',
             cmdlist = [binary_name]
         mlog.debug('Running test binary command: ', mesonlib.join_args(cmdlist))
         try:
-            # fortran code writes to stdout
-            pe = subprocess.run(cmdlist, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+            mlog.debug('Sanity check run command line: ', ' '.join(cmdlist))
+            pe, _, _ = Popen_safe_logged(cmdlist, 'Sanity check', cwd=work_dir)
+            if pe.returncode != 0:
+                raise mesonlib.EnvironmentException(f'Executables created by {self.language} compiler {self.name_string()} are not runnable.')
         except Exception as e:
             raise mesonlib.EnvironmentException(f'Could not invoke sanity test executable: {e!s}.')
-        if pe.returncode != 0:
-            raise mesonlib.EnvironmentException(f'Executables created by {self.language} compiler {self.name_string()} are not runnable.')
```

Comment:
You are moving the pe.returncode exception around I guess in order to make it stack with "Could not invoke"? It's not obvious to me that the message is better now. It is certainly longer though.

Reply:
Dropped.

Reply:
(same here)